lib/src/analysis_options.dart

```diff
@@ -8,92 +8,53 @@ import 'dart:isolate';
 
 import 'package:http/http.dart' as http;
 import 'package:logging/logging.dart';
-import 'package:path/path.dart' as p;
 import 'package:retry/retry.dart';
 import 'package:yaml/yaml.dart' as yaml;
 
 final _logger = Logger('analysis_options');
 
-String? _cachedFlutterOptionsOnGithub;
 String? _cachedLintsCoreInResolvedReferences;
 String? _cachedLintsCoreOptionsOnGithub;
 
-/// Returns the default analysis options (in yaml format).
-Future<String> getDefaultAnalysisOptionsYaml({
-  required bool usesFlutter,
-  required String? flutterSdkDir,
-}) async {
-  if (usesFlutter) {
-    return await _getFlutterAnalysisOptions(flutterSdkDir);
-  } else {
-    return await _getLintsCoreAnalysisOptions();
-  }
-}
-
-Future<String> _getFlutterAnalysisOptions(String? flutterSdkDir) async {
-  // try to load local file
-  flutterSdkDir ??= Platform.environment['FLUTTER_ROOT'];
-  if (flutterSdkDir != null &&
-      flutterSdkDir.isNotEmpty &&
-      await Directory(flutterSdkDir).exists()) {
-    final file = File(p.join(flutterSdkDir, 'packages', 'flutter', 'lib',
-        'analysis_options_user.yaml'));
-    if (await file.exists()) {
-      return await file.readAsString();
-    }
-  }
-
-  // try to load latest from github
-  if (_cachedFlutterOptionsOnGithub != null) {
-    return _cachedFlutterOptionsOnGithub!;
-  }
-  try {
-    final rs = await _httpGetWithRetry(Uri.parse(
-        'https://raw.githubusercontent.com/flutter/flutter/master/packages/flutter/lib/analysis_options_user.yaml'));
-    if (rs.statusCode == 200) {
-      _cachedFlutterOptionsOnGithub = rs.body;
-      return _cachedFlutterOptionsOnGithub!;
-    }
-  } catch (_) {
-    // no-op
-  }
-
-  // fallback empty options
-  _logger.warning('Unable to load default Flutter analysis options.');
-  return '';
-}
+/// The default analysis options configuration (in its raw yaml format).
+Future<String> getDefaultAnalysisOptionsYaml() async =>
+    await _getLintsCoreAnalysisOptions();
 
 Future<String> _getLintsCoreAnalysisOptions() async {
-  // try to load local lints from the resolved package references
-  if (_cachedLintsCoreInResolvedReferences != null) {
-    return _cachedLintsCoreInResolvedReferences!;
+  // Try to load local lints from the resolved package references.
+  if (_cachedLintsCoreInResolvedReferences case final cachedCoreLints?) {
```

Comment:
I don't find this more readable than the `!= null` version. Could you please change it back (also the one below).

Reply:
Fair! The syntax does take a bit of getting used to :)

I reverted this change but then ended up simplifying the implementation a bit further as we ended up only using one of the cached fields anyway. So this private method is now agnostic of the caching. Let me know what you think, thanks!